code/datums/dna.dm

```diff
@@ -406,6 +489,44 @@
 	eye_color = sanitize_hexcolor(getblock(structure, DNA_EYE_COLOR_BLOCK))
 	facial_hairstyle = GLOB.facial_hairstyles_list[deconstruct_block(getblock(structure, DNA_FACIAL_HAIRSTYLE_BLOCK), GLOB.facial_hairstyles_list.len)]
 	hairstyle = GLOB.hairstyles_list[deconstruct_block(getblock(structure, DNA_HAIRSTYLE_BLOCK), GLOB.hairstyles_list.len)]
+	var/features = dna.uni_features
+	if("mcolor" in dna.features)
```

Comment:
This is an associative list.
Searching by index is much faster than doing a linear search.
You can instead do something like `if(dna.features["mcolor"])`

If you plan to use a list like this one extensively, consider putting it in a local variable to avoid so many object accesses each time.